lib/ansible/utils/display.py

```diff
@@ -48,6 +49,50 @@
     # Python 3, we already have raw_input
     pass
 
+# Set the locale to the users default setting
+locale.setlocale(locale.LC_ALL, '')
+
+_LIBC = ctypes.cdll.LoadLibrary(ctypes.util.find_library('c'))
+# Set argtypes, to avoid segfault if the wrong type is provided,
+# restype is assumed to be c_int
+_LIBC.wcwidth.argtypes = (ctypes.c_wchar,)
+_LIBC.wcswidth.argtypes = (ctypes.c_wchar_p, ctypes.c_int)
+# Max for c_int
+_MAX_INT = 2 ** (ctypes.sizeof(ctypes.c_int) * 8 - 1) - 1
+
+
+def get_text_width(text):
```

Comment:
The algorithm in here looks good.  There are some extra corner cases where wcwidth is returning -1 that we might want to handle.  Maybe adapting this test: https://github.com/fedora-infra/kitchen/blob/develop/kitchen3/tests/test_text_display.py#L48 to the unittests for this function and then making it pass.

One thing I note is that the unicode tables are updated when new versions of the unicode standard are released.  Python is generally updated quickly [but not backported] but the various libc implementations are much slower.  So if we rely on libc there will be platforms where wcwidth returns the wrong value.  In previous code I've bundled tables mapping codepoints to width to do better than libc but I'm not sure if we want to do that (For one thing, pure python is slower than wcwidth).  Maybe as a fallback in case libc is extremely broken (like, it doesn't handle hangul, kanji, or devanagari?  Although I'm not sure how we'd check for that.)  I do have code to generate those tables if we want to adopt that strategy.